### PR TITLE
Added workaround for corrupted zip chunks

### PR DESCRIPTION
**Pull request by azhavoro**
### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: changelog.d/20231208_152516_andrey_workarround_cache_corrupted_data.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+### Fixed
+
+- Added workaround for corrupted cached zip chunks
+  (<https://github.com/opencv/cvat/pull/XXXX>)
```
Comment (bsekachev) [inline changelog.d/20231208_152516_andrey_workarround_cache_corrupted_data.md:4]:
```suggestion
  (<https://github.com/opencv/cvat/pull/7243>)
```

File: cvat/apps/engine/cache.py [hunk 2]
```diff
@@ -43,17 +44,29 @@ def __init__(self, dimension=DimensionType.DIM_2D):
...
+            return item
+
+        slogger.glob.info(f'Starting to get chunk from cache: key {key}')
+        item = self._cache.get(key)
+        slogger.glob.info(f'Ending to get chunk from cache: key {key}, is_cached {bool(item)}')
+        if not item:
+            item = create_item()
+        else:
+            # check control sum
+            if item[2] != zlib.crc32(item[0].getbuffer()):
```
[the remark this answers is not in the document]
Comment (azhavoro) [reply cvat/apps/engine/cache.py]: Good catch, fixed

File: cvat/apps/engine/cache.py [hunk 3]
```diff
@@ -43,17 +44,30 @@ def __init__(self, dimension=DimensionType.DIM_2D):
...
+        item = self._cache.get(key)
+        slogger.glob.info(f'Ending to get chunk from cache: key {key}, is_cached {bool(item)}')
+        if not item:
+            item = create_item()
+        else:
+            # compare checksum
+            item_checksum = item[2] if len(item) == 3 else None
+            if item_checksum != zlib.crc32(item[0].getbuffer()):
+                slogger.glob.info(f'Recreating cache item {key} due to checksum mismatch')
+                item = create_item()
```
Comment (bsekachev) [inline cvat/apps/engine/cache.py:68]: I would propose to recreate chunk and compute checksum if checksum does not exist

Like:
```
if not item or len(item) != 3:
    item = create_item()
```

File: cvat/apps/engine/cache.py [hunk 4]
```diff
@@ -43,17 +44,31 @@ def __init__(self, dimension=DimensionType.DIM_2D):
...
+            return item
+
+        slogger.glob.info(f'Starting to get chunk from cache: key {key}')
+        item = self._cache.get(key)
+        slogger.glob.info(f'Ending to get chunk from cache: key {key}, is_cached {bool(item)}')
+        if not item:
+            item = create_item()
+        else:
+            # compare checksum
+            item_data = item[0].getbuffer() if isinstance(item[0], io.BytesIO) else io.BytesIO(item[0])
```
Comment (SpecLad) [inline cvat/apps/engine/cache.py:65]:
```suggestion
            item_data = item[0].getbuffer() if isinstance(item[0], io.BytesIO) else item[0]
```

**Review by SpecLad — APPROVED** [review]

**Pull request merged**